### PR TITLE
Update voltdb client to version 8.3

### PR DESCRIPTION
**Pull request by vianel**
- Updated voltdbCliend dependency to version 8.3

- Updated updateApplicationCatalogAsync function to call procedure @UpdateApplicationCatalog using call procedure 

- Added WriteSummaryCSV function

- Added SuccessCallback to getBulkLoader function

### CODE REVIEW

File: src/main/scala/com/full360/voltdbscala/Client.scala [hunk 1]
```diff
@@ -133,12 +131,12 @@ trait Client {
    * @param maxBatchSize Batch size to collect for the table before pushing a bulk insert.
    * @param upsert       set to true if want upsert instead of insert
    * @param f            Function invoked by the BulkLoaderFailureCallBack used for notification of failed inserts.
+   * @param f2           Function invoked by BulkLoaderSuccessCallback used for notifications on successful load operations
    * @return instance of VoltBulkLoader
    * @throws Exception if tableName can't be found in the catalog.
    */
-  def getNewBulkLoader(tableName: String, maxBatchSize: Int, upsert: Boolean = false)(f: (Any, Seq[AnyRef], ClientResponse) ⇒ Unit): VoltBulkLoader =
```
Comment (albertogg) [inline src/main/scala/com/full360/voltdbscala/Client.scala:139]: I think this method should not be removed. It exist in two variants with method overloading.

File: src/main/scala/com/full360/voltdbscala/Client.scala [hunk 2]
```diff
@@ -110,16 +110,14 @@ trait Client {
...
    * [[com.full360.voltdbscala.ProcedureNotQueuedException ProcCallException]] if queueing did not take place.</p>
    *
-   * @see [[org.voltdb.client.Client#updateApplicationCatalog(org.voltdb.client.ProcedureCallback, java.io.File, java.io.File) Client.updateApplicationCatalog]]
-   *
    * @param catalogPath    Path to the catalog jar file.
    * @param deploymentPath Path to the deployment file.
    * @return a scala Future holding an instance of a [[org.voltdb.client.ClientResponse ClientResponse]] or exception.
    */
-  def updateApplicationCatalogAsync(catalogPath: File, deploymentPath: File)(implicit ec: ExecutionContext): Future[ClientResponse] =
+  def updateApplicationCatalogAsync(catalogPath: String, deploymentPath: String)(implicit ec: ExecutionContext): Future[ClientResponse] =
```
Comment (albertogg) [inline src/main/scala/com/full360/voltdbscala/Client.scala]: This method should be removed as it was deprecated. Please, remove it.

File: project/ProjectSettings.scala [hunk 3]
```diff
@@ -59,7 +59,6 @@ object ProjectSettings {
     "-Ywarn-dead-code",
     "-Ywarn-numeric-widen",
     "-Xfuture",
-    "-Xlint:-unused",
```
Comment (albertogg) [inline project/ProjectSettings.scala]: Why was this removed?

Comment (vianel) [reply project/ProjectSettings.scala]: `[error] '-unused' is not a valid choice for '-Xlint'`

**Comment by albertogg** [comment]
From what I see, we are missing some functionality that should be easy to add:
- updateClasses (sync version)

**Comment by albertogg** [comment]
Some functionality missing that's not that easy to add:
- createStatsContext

This method introduces some type that we are not currently handling `ClientStatsContext`

**Comment by albertogg** [comment]
> Some functionality missing that's not that easy to add:
> 
> * createStatsContext
> 
> This method introduces some type that we are not currently handling `ClientStatsContext`

We might not need this.

**Comment by albertogg** [comment]
hey @vianel it's not passing tests in Scala 2.12

**Review by igoticecream — COMMENTED** [review]
Looks good

**Comment by albertogg** [comment]
hey, @vianel, let's update the current snapshot version in this branch to `0.8.0-SNAPSHOT` so when we merge to master we get our new SNAPSHOT release
